File: cutinline.py
queue = [input().strip() for i in range(int(input()))]

for j in range(int(input())):
    event = input().split()
    if event[0] == 'cut':
        for k in range(len(queue)):
            if queue[k] == event[2]:
                queue = queue[:k] + [event[1]] + queue[k:]
                break
    elif event[0] == 'leave':
        queue.remove(event[1])

for person in queue:
    print(person)

File: cutinline.java
import java.util.*;
import java.io.*;

public class cutinline {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        ArrayList<String> queue = new ArrayList<String>();
        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) queue.add(br.readLine().strip());
        int Q = Integer.parseInt(br.readLine());
        for (int i = 0; i < Q; i++) {
            String[] line = br.readLine().split(" ");
            if (line[0].equals("cut")) queue.add(queue.indexOf(line[2]),line[1]);
            else queue.remove(line[1]);
        }
        for (String s: queue) pw.println(s);
        br.close();
        pw.close();
    }
}